src/utils.ts

```diff
@@ -130,11 +130,19 @@ export function mockGlobalTSConfigSchema(globals: any) {
     { __TS_CONFIG__: config };
 }
 
+const tsConfigCache: { [key: string]: any } = {};
 export function getTSConfig(globals, collectCoverage: boolean = false) {
   let config = getTSConfigOptionFromConfig(globals);
   const skipBabel = getTSJestConfig(globals).skipBabel;
   const isReferencedExternalFile = typeof config === 'string';
 
+  // check cache before resolving configuration
+  // NB: config is a string unless taken from __TS_CONFIG__, which should be immutable (and is deprecated anyways)
+  const tsConfigCacheKey = JSON.stringify([skipBabel, collectCoverage, isReferencedExternalFile ? config : undefined]);
```

Comment:
Ah, I thought you meant that you wanted a comment here on GitHub. I've just added the explanation in the code.